pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -607,6 +605,13 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 	if vmSnapshotCpy.DeletionTimestamp != nil {
 		// go into error state
 		if vmSnapshotProgressing(vmSnapshotCpy) {
+			if source != nil {
```

Comment:
I think it is better to do this code in the main updatesnaphot before the vmSnapshotCpy.DeletionTimestamp != nil, since it will cover also the deadline, in both cases we mark error on the snapshot which makes vmSnapshotProgressing(vmSnapshotCpy) to be false and enter the if in the reconcile loop

Reply:
was thinking we may actually want to handle deadline in the snapshotcontent since that is where we freeze in the first place.  what do you think?

Reply:
I dont think so, it is relevant for every snapshot not only with freeze.. also it is relevant for the snapshot as a whole, we compare the timeout to the time the snapshot was created

Reply:
Good call